ic-os/components/BUILD.bazel

```diff
@@ -1,51 +1,53 @@
-# Boundary GuestOS
-exports_files(
-    glob(["boundary-guestos/**"]),
-    ["//ic-os/boundary-guestos:__subpackages__"],
-)
+load("//ic-os/components/conformance-tests:defs.bzl", "check_unused_components_test")
+load("boundary-guestos.bzl", boundaryos_component_files = "component_files")
+load("guestos.bzl", guestos_component_files = "component_files")
+load("hostos.bzl", hostos_component_files = "component_files")
+load("setupos.bzl", setupos_component_files = "component_files")
 
-# GuestOS
 PUBLIC_GUESTOS_EXPORTS = [
     "ic/ic.json5.template",
     "networking/dev-certs/canister_http_test_ca.cert",
     "networking/dev-certs/canister_http_test_ca.key",
     "hostos-scripts/build-bootstrap-config-image.sh",
 ]
 
-exports_files(
-    glob(["guestos/**"]),
-    ["//ic-os/guestos:__subpackages__"],
-)
-
 exports_files(PUBLIC_GUESTOS_EXPORTS)
 
-# HostOS
-exports_files(
-    glob(["hostos/**"]),
-    ["//ic-os/hostos:__subpackages__"],
-)
-
-# SetupOS
-exports_files(
-    glob(["setupos/**"]),
-    ["//ic-os/setupos:__subpackages__"],
-)
-
-# Common
 exports_files(
     glob(
         ["**"],
-        exclude = [
-            "boundary-guestos/**",
-            "guestos/**",
-            "hostos/**",
-            "setupos/**",
-        ] + PUBLIC_GUESTOS_EXPORTS,
+        exclude = PUBLIC_GUESTOS_EXPORTS,
     ),
-    [
+    visibility = [
         "//ic-os/boundary-guestos:__subpackages__",
         "//ic-os/guestos:__subpackages__",
         "//ic-os/hostos:__subpackages__",
         "//ic-os/setupos:__subpackages__",
     ],
 )
+
+used_components = dict()
+
+used_components.update(guestos_component_files)
```

Comment:
optional nit: you can write
`used_components = guestos_component_files | hostos_component_files | setupos_component_files | boundaryos_component_files`